common/elasticsearch/client.go

```diff
@@ -43,9 +47,26 @@ type (
 		Scroll(ctx context.Context, scrollID string) (*elastic.SearchResult, ScrollService, error)
 		ScrollFirstPage(ctx context.Context, index, query string) (*elastic.SearchResult, ScrollService, error)
 		Count(ctx context.Context, index, query string) (int64, error)
-		RunBulkProcessor(ctx context.Context, p *BulkProcessorParameters) (*elastic.BulkProcessor, error)
+		RunBulkProcessor(ctx context.Context, p *BulkProcessorParameters) (BulkProcessor, error)
 		PutMapping(ctx context.Context, index, root, key, valueType string) error
-		CreateIndex(ctx context.Context, index string) error
+		CreateIndex(ctx context.Context, index string) (bool, error)
+
+		IsNotFoundError(err error) bool
```

Comment:
consider making the es layer do the conversion of error

Reply:
This is very good point! Fixed.